packages/ragbits-core/src/ragbits/core/vector_store/chromadb_store.py

```diff
@@ -61,114 +70,62 @@ def from_config(cls, config: dict) -> ChromaDBStore:
-        embedding = entry.vector
-
-        metadata = {
-            "__key": entry.key,
-            "__metadata": json.dumps(entry.metadata, default=str),
-        }
-
-        return doc_id, embedding, metadata
-
-    @property
-    def embedding_function(self) -> Embeddings | chromadb.EmbeddingFunction:
-        """
-        Returns the embedding function.
-
-        Returns:
-            The embedding function.
-        """
-        return self._embedding_function
-
     async def store(self, entries: list[VectorDBEntry]) -> None:
         """
         Stores entries in the ChromaDB collection.
 
         Args:
             entries: The entries to store.
         """
-        entries_processed = list(map(self._process_db_entry, entries))
-        ids, embeddings, metadatas = map(list, zip(*entries_processed, strict=False))
+        # TODO: Think about better id components for hashing
+        ids = [sha256(entry.key.encode("utf-8")).hexdigest() for entry in entries]
+        embeddings = [entry.vector for entry in entries]
+        metadatas = [
+            {
+                "__key": entry.key,
+                "__metadata": json.dumps(entry.metadata, default=str),
+            }
+            for entry in entries
+        ]
+        self._collection.add(ids=ids, embeddings=embeddings, metadatas=metadatas)  # type: ignore
 
-        self._collection.add(ids=ids, embeddings=embeddings, metadatas=metadatas)
-
-    async def retrieve(self, vector: list[float], k: int = 5) -> list[VectorDBEntry]:
+    async def retrieve(self, vector: list[float], options: VectorStoreOptions | None = None) -> list[VectorDBEntry]:
         """
         Retrieves entries from the ChromaDB collection.
 
         Args:
             vector: The vector to query.
-            k: The number of entries to retrieve.
+            options: The options for querying the vector store.
 
         Returns:
             The retrieved entries.
         """
-        query_result = self._collection.query(query_embeddings=vector, n_results=k, include=["metadatas", "embeddings"])
-        metadatas = query_result.get("metadatas") or []
-        embeddings = query_result.get("embeddings") or []
-
-        db_entries = []
-        for meta_list, embeddings_list in zip(metadatas, embeddings, strict=False):
-            for meta, embedding in zip(meta_list, embeddings_list, strict=False):
-                db_entry = VectorDBEntry(
-                    key=str(meta["__key"]),
-                    vector=list(embedding),
-                    metadata=json.loads(str(meta["__metadata"])),
-                )
-
-                db_entries.append(db_entry)
+        options = self._default_options if options is None else options
+        results = self._collection.query(
+            query_embeddings=vector,
+            n_results=options.k,
+            include=["metadatas", "embeddings", "distances"],
+        )
+        metadatas = results.get("metadatas") or []
```

Comment:
```suggestion
        metadatas = results.get("metadatas", [])
```

Reply:
I was thinking about this, but I couldn't infer proper types with this, cuz `get` always returns `Optional`

Reply:
ok, so let's leave it as it is